pkg/bosh/manifest/dns.go

```diff
@@ -0,0 +1,323 @@
+		}
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, c, &configMap, configMapMutateFn(&configMap)); err != nil {
+		return err
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, c, &deployment, deploymentMapMutateFn(&deployment)); err != nil {
+		return err
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, c, &service, mutate.ServiceMutateFn(&service)); err != nil {
+		return err
+	}
+
+	dns.localDNSIP = service.Spec.ClusterIP
+	return nil
+}
+
+var re = regexp.MustCompile(`nameserver\s+([\d\.]*)`)
+
+func (dns *boshDomainNameService) rootDNSIP() string {
+	content, err := ioutil.ReadFile("/etc/resolv.conf")
+	if err != nil {
+		content = []byte{}
+	}
+	return GetNameserverFromResolveConfig(content)
+}
+
+// GetNameserverFromResolveConfig read nameserver from resolve.conf
+func GetNameserverFromResolveConfig(content []byte) string {
+	found := re.FindSubmatch(content)
+	if len(found) == 0 {
+		ctxlog.Errorf(context.TODO(), "No nameserver found in resolve.conf using 1.1.1.1")
+		return "1.1.1.1"
+	}
+	return string(found[1])
+
+}
+
+type simpleDomainNameService struct {
+}
+
+// FindServiceNames see interface
+func (dns *simpleDomainNameService) FindServiceNames(instanceGroupName string, deploymentName string) []string {
+	return []string{serviceName(instanceGroupName, deploymentName, 63)}
+}
+
+// HeadlessServiceName see interface
+func (dns *simpleDomainNameService) HeadlessServiceName(instanceGroupName string, deploymentName string) string {
+	return serviceName(instanceGroupName, deploymentName, 63)
+}
+
+// DNSSetting see interface
+func (dns *simpleDomainNameService) DNSSetting() (corev1.DNSPolicy, *corev1.PodDNSConfig) {
+	return corev1.DNSDefault, nil
+}
+
+// Reconcile see interface
+func (dns *simpleDomainNameService) Reconcile(ctx context.Context, c client.Client, setOwner func(object metav1.Object) error) error {
+	return nil
+}
+
+func serviceName(instanceGroupName string, deploymentName string, maxLength int) string {
+	serviceName := fmt.Sprintf("%s-%s", deploymentName, names.Sanitize(instanceGroupName))
+	if len(serviceName) > maxLength {
+		sumHex := md5.Sum([]byte(serviceName))
+		sum := hex.EncodeToString(sumHex[:])
+		serviceName = fmt.Sprintf("%s-%s", serviceName[:maxLength-len(sum)-1], sum)
+	}
+	return serviceName
+}
+
+func configMapMutateFn(configMap *corev1.ConfigMap) controllerutil.MutateFn {
+	updated := configMap.DeepCopy()
+	return func() error {
+		configMap.Labels = updated.Labels
+		configMap.Annotations = updated.Annotations
+		configMap.Data = updated.Data
+		return nil
+	}
+}
+
+func deploymentMapMutateFn(deployment *appsv1.Deployment) controllerutil.MutateFn {
+	updated := deployment.DeepCopy()
+	return func() error {
+		deployment.Labels = updated.Labels
+		deployment.Annotations = updated.Annotations
+		deployment.Spec = updated.Spec
+		return nil
+	}
+}
+
+const corefile = `
+.:8053 {
+  health
+  %s
+  rewrite name substring service.cf.internal %s.svc.cluster.local
+  forward . %s
```

Comment:
This should be only `forward . /etc/resolv.conf`. The CoreDNS pod should use the upstream DNS set by k8s. In this case, you won't need 2 other functions above that parse `/etc/resolv.conf` on the cf-operator.

Reply:
We tried this, but it was not working out of the box. Perhaps, we should give it a second try.

Reply:
We are using `forward . /etc/resolv.conf` now.